### ISSUE TITLE
Error while using `cuda` function (cudawarping::resize)

### ISSUE THREAD

**Issue by kahlil29** [twistedfall/opencv-rust#175]
Before reporting an issue please first check the
[troubleshooting guide](https://github.com/twistedfall/opencv-rust/blob/master/README.md#troubleshooting). If
the issue you're encountering is not solved thereby please state the following in your bugreport:
1. Operating system : Ubuntu (Linux)
2. The way you installed OpenCV: Manual compilation with the following options : 
```
cmake -D CMAKE_BUILD_TYPE=RELEASE     -D CMAKE_INSTALL_PREFIX=/usr/local     -D WITH_CUDA=ON     -D CUDA_ARCH_BIN=6.1     -D ENABLE_FAST_MATH=1     -D CUDA_FAST_MATH=1     -D WITH_CUBLAS=1     -D INSTALL_C_EXAMPLES=ON     -D INSTALL_PYTHON_EXAMPLES=ON     -D OPENCV_GENERATE_PKGCONFIG=ON     -D OPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules     -D BUILD_EXAMPLES=ON ..
```

3. OpenCV version : `4.2.0`
4. rustc version (`rustc --version`) : `rustc 1.47.0 (18bf6b4f0 2020-10-07)`
5. Attach the full output of the following command from your project directory:
   ```shell script
   RUST_BACKTRACE=full cargo build -vv 
   ```
Note: Build succeeded so did a `cargo run` version of the above command 

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: -217, message: "OpenCV(4.2.0) /home/ubuntu/source_repos/opencv_build/opencv_contrib/modules/cudawarping/src/cuda/resize.cu:265: error: (-217:Gpu API call) no kernel image is available for execution on the device in function \'call_resize_cubic_glob\'\n" }', src/img_utils.rs:24:6
stack backtrace:
   0:     0x557145d29a20 - std::backtrace_rs::backtrace::libunwind::trace::h577ea05e9ca4629a
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/../../backtrace/src/backtrace/libunwind.rs:96
   1:     0x557145d29a20 - std::backtrace_rs::backtrace::trace_unsynchronized::h50b9b72b84c7dd56
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:     0x557145d29a20 - std::sys_common::backtrace::_print_fmt::h6541cf9823837fac
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:79
   3:     0x557145d29a20 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hf64fbff071026df5
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:58
   4:     0x557145d42e0c - core::fmt::write::h9ddafa4860d8adff
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/fmt/mod.rs:1082
   5:     0x557145d27e87 - std::io::Write::write_fmt::h1d2ee292d2b65481
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/io/mod.rs:1514
   6:     0x557145d2baf0 - std::sys_common::backtrace::_print::ha25f9ff5080d886d
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:61
   7:     0x557145d2baf0 - std::sys_common::backtrace::print::h213e8aa8dc5405c0
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:48
   8:     0x557145d2baf0 - std::panicking::default_hook::{{closure}}::h6482fae49ef9d963
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:200
   9:     0x557145d2b83c - std::panicking::default_hook::he30ad7589e0970f9
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:219
  10:     0x557145d2c153 - std::panicking::rust_panic_with_hook::haa1ed36ada4ffb03
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:569
  11:     0x557145d2bd29 - std::panicking::begin_panic_handler::{{closure}}::h7001af1bb21aeaeb
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:476
  12:     0x557145d29eac - std::sys_common::backtrace::__rust_end_short_backtrace::h39910f557f5f2367
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:153
  13:     0x557145d2bce9 - rust_begin_unwind
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:475
  14:     0x557145d42131 - core::panicking::panic_fmt::h4e2659771ebc78eb
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/panicking.rs:85
  15:     0x557145d41f53 - core::option::expect_none_failed::h448b58a024c2c33a
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/option.rs:1221
  16:     0x557145ccc1ea - core::result::Result<T,E>::unwrap::h7ba516a5339a6f61
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/result.rs:973
  17:     0x557145cccf96 - img_utils::main::h865baefc712d3952
                               at /home/ubuntu/rust-foodScoring-service/src/img_utils.rs:20
  18:     0x557145ccc4db - core::ops::function::FnOnce::call_once::hed74266a605fbd3b
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/ops/function.rs:227
  19:     0x557145ccbe8e - std::sys_common::backtrace::__rust_begin_short_backtrace::h5cde7955512c0e68
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:137
  20:     0x557145ccce31 - std::rt::lang_start::{{closure}}::he39143d797387cd6
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/rt.rs:66
  21:     0x557145d2c551 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h6a3209f124be2235
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/ops/function.rs:259
  22:     0x557145d2c551 - std::panicking::try::do_call::h88ce358792b64df0
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:373
  23:     0x557145d2c551 - std::panicking::try::h6311c259678e50fc
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:337
  24:     0x557145d2c551 - std::panic::catch_unwind::h56c5716807d659a1
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panic.rs:379
  25:     0x557145d2c551 - std::rt::lang_start_internal::h73711f37ecfcb277
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/rt.rs:51
  26:     0x557145ccce07 - std::rt::lang_start::he63713b6adabd766
                               at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/rt.rs:65
  27:     0x557145ccdd7a - main
  28:     0x7f196c59fb97 - __libc_start_main
  29:     0x557145ccbd2a - _start
  30:                0x0 - <unknown>
```

I googled this error and it seems common in cases where the CUDA compute capability is less than 3.5 (very common issue with PyTorch it seems). 
Here's the output of `print_cuda_device_info(0)` : 
```
*** CUDA Device Query (Runtime API) version (CUDART static linking) ***

Device count: 1

Device 0: "Tesla T4"
  CUDA Driver Version / Runtime Version          10.20 / 10.0
  CUDA Capability Major/Minor version number:    7.5
  Total amount of global memory:                 15110 MBytes (15843721216 bytes)
  GPU Clock Speed:                               1.59 GHz
  Max Texture Dimension Size (x,y,z)             1D=(131072), 2D=(131072,65536), 3D=(16384,16384,16384)
  Max Layered Texture Size (dim) x layers        1D=(32768) x 2048, 2D=(32768,32768) x 2048
  Total amount of constant memory:               65536 bytes
  Total amount of shared memory per block:       49152 bytes
  Total number of registers available per block: 65536
  Warp size:                                     32
  Maximum number of threads per block:           1024
  Maximum sizes of each dimension of a block:    1024 x 1024 x 64
  Maximum sizes of each dimension of a grid:     2147483647 x 65535 x 65535
  Maximum memory pitch:                          2147483647 bytes
  Texture alignment:                             512 bytes
  Concurrent copy and execution:                 Yes with 3 copy engine(s)
  Run time limit on kernels:                     No
  Integrated GPU sharing Host Memory:            No
  Support host page-locked memory mapping:       Yes
  Concurrent kernel execution:                   Yes
  Alignment requirement for Surfaces:            Yes
  Device has ECC support enabled:                Yes
  Device is using TCC driver mode:               No
  Device supports Unified Addressing (UVA):      Yes
  Device PCI Bus ID / PCI location ID:           0 / 30
  Compute Mode:
      Default (multiple host threads can use ::cudaSetDevice() with device simultaneously)

deviceQuery, CUDA Driver = CUDART, CUDA Driver Version  = 10.20, CUDA Runtime Version = 10.0, NumDevs = 1
```

**Comment by kahlil29** [comment]
Here's the code where the actual call to `cudawarping::resize` is made: 

```
    let mut resizedAndProcessedImageMat = GpuMat::default()?;

    let imgMatrix = imgcodecs::imread(&currentImgPath, imgcodecs::IMREAD_COLOR)?;

    let mut img_gpu = GpuMat::default()?;
    img_gpu.upload(&imgMatrix)?;
    let dimensions = Size_::new(512, 512);

    match cudawarping::resize(
        &img_gpu,
        &mut resizedAndProcessedImageMat,
        dimensions,
        0.0,
        0.0,
        2,
        &mut opencv::core::Stream::null()? 
    ) {
        Ok(()) => {
            let mut newVec = opencv::types::VectorOfi32::new();
            newVec.push(IMWRITE_PNG_COMPRESSION);
            newVec.push(0);

            let mut encodedImgBytes = opencv::core::Vector::new();
            imgcodecs::imencode(
                ".png",
                &resizedAndProcessedImageMat,
                &mut encodedImgBytes,
                &opencv::types::VectorOfi32::new(),
            )?;
            match std::fs::write(currentImgPath, encodedImgBytes) {
                Ok(()) => return Ok(()),
                Err(fileWriteErr) => {
                    return Err(opencv::Error::new(
                        66, // random code 
                        format!(
                            "Error while writing pre-processed & resized image to file: {}",
                            fileWriteErr
                        ),
                    ))
                }
            }
        }
        Err(opencvResizeErr) => return Err(opencvResizeErr),
    };
```

**Comment by kahlil29** [comment]
Okay so this is not a `opencv-rust` issue per se, but just posed it here incase anyone has encountered it before. 
I did some digging and it seems that this is caused by issues with the `CUDA_ARCH_BIN` & GPU Compute Capability version mis-match. 

My GPU has Compute Capability `7.5`
But I've used `CUDA_ARCH_BIN=6.1` which seems to me like the the root cause of the error. 
Will try building OpenCV again with `CUDA_ARCH_BIN=7.5`
